jax/core.py

```diff
@@ -2122,12 +2122,16 @@ def pp_eqn(eqn, context: JaxprPpContext, *, print_shapes=True, source_info=False
   lhs = pp_vars(eqn.outvars, context, print_shapes=print_shapes)
   annotation = (source_info_util.summarize(eqn.source_info)
                 if source_info else None)
-  return pp.concat([
-    lhs, pp.text(" = ", annotation=annotation), pp.text(eqn.primitive.name),
-    pp_kv_pairs(sorted(eqn.params.items()), context),
-    pp.text(" ") + pp_vars(eqn.invars, context)
-  ])
-
+  rule = pp_eqn_rules.get(eqn.primitive)
+  if rule:
```

Comment:
I would be tempted to add a flag to turn off special pretty-printing to the printer. It may be useful sometimes to see the unperturbed generic form.

Reply:
Good idea.

Reply:
Done.